src/lib.rs

```diff
@@ -82,7 +82,28 @@
 //!
 //! `os-arch` and `authors` also can't be overriden and are derived from the environment.
 
-#![warn(clippy::unneeded_field_pattern, clippy::match_bool, clippy::get_unwrap)]
+#![warn(
```

Comment:
I wonder why we would need custom clippy configuration, and not use the reasonable defaults? 
Personal prefectures?

Reply:
Well, as I see it - @MarcoIeni showed us that Clippy could actually spot quite a few cases that are considered good Rust practice - it just doesn't do it by default. Why it doesn't, I don't know.

Apart from the `.gitignore` file (which came about because I've been trying out commitlint in githooks to improve the commit messages), I actually think it's a shame these warnings are not enabled by default.

This PR is not exactly a must, but it does show that Clippy can help us even more, though it does require us to "opt in".
If nothing else, we should at least consider enabling some of them.

Reply:
I think they are pretty neat. I tend to like them actually 🙂